startup-project/app/scripts/services/spotifyservice.js

```diff
@@ -0,0 +1,298 @@
+'use strict';
+
+/**
+ * @ngdoc service
+ * @name startupProjectApp.SpotifyService
+ * @description
+ * # SpotifyService
+ * Factory in the startupProjectApp.
+ */
+angular.module('startupProjectApp')
+  .factory('SpotifyService', [
+    'store',
+    '$q',
+    '$http',
+    function (store, $q, $http) {
+
+    //config for spotifyService
+    this.config = {};
+      this.config.clientId = null;
+      this.config.redirectUri = null;
```

Comment:
Why you do not define all this in the line where you initialized the config object?